### ISSUE TITLE
Add correct aria roles to checkbox 

### ISSUE THREAD

**Issue by alisonjoseph** [carbon-design-system/carbon-components-react#469]
The vanilla checkbox has the correct `aria-checked` values, but this is missing in the React version.

**Comment by alisonjoseph** [comment]
Closing this, since we are using the correct semantic html the aria roles are not needed. 

https://www.w3.org/TR/html5/dom.html#aria-usage-note
`In the majority of cases setting an ARIA role and/or aria-* attribute that matches the default implicit ARIA semantics is unnecessary and not recommended as these properties are already set by the browser.`
